<launch>
  <arg name="camera" default="cam_pcl_1"/>
  
  <node pkg="rs-pcl2" type="diff" name="diff" output="screen">
  <!-- Step 1 - CUBE SEGMENTATION  -->
    <remap from="diff/input" to="/$(arg camera)/depth/color/points" />
    <remap from="cube/output" to="/$(arg camera)/differential/ladybug/cube" />
    
    <!-- Filter Params for Y axis -->
    <param name="filter_low_y" value="-0.05" />
    <param name="filter_high_y" value="0.06" /> 
    <!-- Filter Params for Z axis -->
    <param name="filter_low_z" value="0" />
    <param name="filter_high_z" value="1" />
    <!-- Filter Params for X axis -->
    <param name="filter_low_x" value="-0.5" />
    <param name="filter_high_x" value="0.5" />


  <!-- Step 2 - DBSCAN Cluster -->
    <remap from="dbscan/input" to="/$(arg camera)/depth/color/points" />
    <!-- DBSCAN Params -->
    <param name="cluster_tolerance" value="0.01" />
    <param name="min_cluster_size" value="100" />
    <param name="max_cluster_size" value="5000" />

    <!-- Marker Array Params -->
    <param name="line_thickness" value="0.005" />
  </node>


</launch>
